### ISSUE TITLE
error: internal compiler error: src\librustc_middle\ich\impls_ty.rs:94: StableHasher: unexpected region '_#1050r

### ISSUE THREAD

**Issue by fragrans** [rust-lang/rust#73204]
I found this error, and hope it might help improve the compiler;
here is the repository;

[https://github.com/fragrans/ice_2020.git](https://github.com/fragrans/ice_2020.git)

```pwsh
E:\bitbucket.org\ice [orphan ≡]> cargo build
   Compiling avc_dmo v0.1.0 (E:\bitbucket.org\ice)
warning: unused imports: `Future`, `TryFuture`
 --> src\main.rs:6:23
  |
6 | use futures::future::{Future, TryFuture, FutureExt};
  |                       ^^^^^^  ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error: internal compiler error: src\librustc_middle\ich\impls_ty.rs:94: StableHasher: unexpected region '_#1050r

thread 'rustc' panicked at 'Box<Any>', src\librustc_errors\lib.rs:907:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.45.0-nightly (5fd2f06e9 2020-05-31) running on x86_64-pc-windows-msvc

note: compiler flags: -C embed-bitcode=no -C debuginfo=2 -C incremental --crate-type bin

note: some of the compiler flags provided by cargo are hidden

error: aborting due to previous error; 1 warning emitted

error: could not compile `avc_dmo`.

To learn more, run the command again with --verbose.
```

**Comment by jonas-schievink** [comment]
Looks like https://github.com/rust-lang/rust/issues/72766. Can you retry on the newest nightly?

**Comment by fragrans** [comment]
Yes, It's already been fixed in rustc 1.46.0-nightly (feb3536eb 2020-06-09) .
Sorry for the noise;
